src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -564,6 +564,7 @@ internal void SendResponse(Message request, Response response)
         {
             // create the response
             var message = request.CreateResponseMessage();
+            if(!request.DisposeScheduled) request.Dispose();
```

Comment:
Yes, I'm sure that its right condition, the purpose of <code>DisposeScheduled</code> property is explained at https://github.com/dotnet/orleans/pull/1946#discussion_r71042599